src/main/java/com/nutomic/syncthingandroid/syncthing/EventProcessor.java

```diff
@@ -116,7 +117,8 @@ public void onEvent(long id, String type, JSONObject data) throws JSONException
                         .putExtra(SettingsActivity.EXTRA_IS_CREATE, true)
                         .putExtra(FolderFragment.EXTRA_DEVICE_ID, deviceId)
                         .putExtra(FolderFragment.EXTRA_FOLDER_ID, folderId);
-                pi = PendingIntent.getActivity(mContext, 0, intent, 0);
+                pi = PendingIntent.getActivity(mContext, (deviceId + folderId).hashCode(),
+                        intent, 0);
```

Comment:
Thanks, I didn't even know that was necessary. However, using a hashcode is definitely a hack. So can you add a comment similar to the one in line 199?